### PR TITLE
chore: move opam files to opam/ directory

### PR DESCRIPTION
**Pull request by rgrinberg**
Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

### CODE REVIEW

**Review by anmonteiro — APPROVED** [review]

**Comment by anmonteiro** [comment]
- I think you'll need to tweak some CI commands that point to relative opam files in source
- I wonder if there's a bug in the install rules, check this failure: https://github.com/ocaml/dune/actions/runs/4613187638/jobs/8154917698?pr=7490
- you need to format the `opam/dune` file

**Comment by rgrinberg** [comment]
@emillon is this problematic for dune-release?

**Comment by emillon** [comment]
Yes, I don't think that `dune-release` supports that at the moment. @Leonidas-from-XIV might have a more precise answer.

**Comment by Leonidas-from-XIV** [comment]
Until recently I didn't know OPAM supported files in `opam/` directly and according to my cursory exploration of the `dune-release` source I found a lot of interesting stuff but no support for `.opam` files in `opam/`.

But it shouldn't be too hard to add that support, it's basically just adding a third option for lookup (`opam`, `<pkg>.opam`, `opam/<pkg>.opam`) and given dune supports generation in `opam/` folders now it would make sense to support that too if people start using that feature.

**Comment by emillon** [comment]
thanks. in the context of dune, we can't merge this PR until dune-release has support for this then.

**Comment by Leonidas-from-XIV** [comment]
@emillon I've created https://github.com/tarides/dune-release/pull/466 but at the moment it is a chicken and egg problem, because it needs dune 3.8 to pass the test.

File: dune-project [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 (strict_package_deps false)
 
+(opam_file_location inside_opam_directory)
```
Comment (Alizter) [inline dune-project:13]:
```suggestion
; .opam files get generated in opam/
(opam_file_location inside_opam_directory)
```

**Comment by emillon** [comment]
I tested that when releasing dune.3.9.0~alpha1 with dune-release.2.0.0 and it worked.

**Comment by Alizter** [comment]
ping on this

**Comment by Alizter** [comment]
@Leonidas-from-XIV there are some CI failures which we got stuck on. In principle , everybody was happy with this.

**Comment by rgrinberg** [comment]
Ali finished it off and it's been merged.

**Pull request closed without merging**